libp2p/switch.nim

```diff
@@ -332,6 +334,9 @@ proc stop*(s: Switch) {.public, async: (raises: [CancelledError]).} =
     except CatchableError as exc:
       warn "error cleaning up transports", description = exc.msg
 
+  if not s.autotls.isNil:
+    await s.autotls.stop()
```

Comment:
Wdyt of modeling AutoTLSManager as a service, so it's setup in line 374? The advantage of it is that Switch wouldnt need to get an additional attribute, the peerInfo can be extracted during the call to `setup`, and you wouldnt have to do this s.autotls.isNil check here and in the stop. 

Can of course be done on a separate PR, but for sure it would look more correct. Then withAutoTLS what it'd do is append autoTLS service to b.services in the SwitchBuilder

Reply:
Neat!